### PR TITLE
Feature/catalogue to entry sets

### PR DESCRIPTION
**Pull request by mindblight**
All changes don't modify code (besides import statements) except for:

* `CatalogComponent.js`
* https://github.com/kryptnostic/gallery/commit/a8e9ac3ae3423258988237a501cd82d23fc24060
* Build system changes to support css `@imports`. (See http://stackoverflow.com/questions/41512707/webpack-css-loader-does-not-resolve-import-ed-css-files-into-js-object)

I've pushed two conceptual changes for feedback:

1. Container folders should be lower-case
2. Files containing components should be named `FooComponent`. This prevents name collisions between components and data. For example, `EntitySetListComponent` would render `EntitySetList`s.

Let me know you're thoughts.

### CODE REVIEW

File: postcss.config.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  "parse": "sugarss",
```
Comment (UnsungHero97) [inline postcss.config.js]: what's `sugarss`?

File: src/containers/catalog/CatalogComponent.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React, { PropTypes } from 'react';
+import { EntitySetList } from './components/EntitySetList';
+import AuthService from '../../utils/AuthService';
+
+
+export class CatalogComponent extends React.Component {
+  static propTypes = {
```
Comment (UnsungHero97) [inline src/containers/catalog/CatalogComponent.js]: I've been following this style...

```
class Organizations extends React.Component {
  ...
}

Organizations.propTypes = {
  ...
};
```

... but I don't really have a solid case for why, just what I've seen in tutorials. Any ideas as to which approach is preferred, using `static` inside the class, or setting it on the class outside?

File: src/containers/catalog/CatalogComponent.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    auth: PropTypes.instanceOf(AuthService)
+  }
+
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (<div>
```
Comment (UnsungHero97) [inline src/containers/catalog/CatalogComponent.js]:
```
return (
  <div>
    ...
  </div>
);
```

File: src/views/Main/components/Navbar.js [hunk 4]
```diff
@@ -40,58 +40,45 @@ export class Navbar extends React.Component {
...
   render() {
     return (
       <div className={styles.navbarContainer}>
-        <button
-          className={this.getButtonClass(PageConsts.HOME)}
-          onClick={() => {
-            this.updateState(PageConsts.HOME);
-          }}
-        >
+        <Link to={`/${PageConsts.HOME}`}
```
Comment (UnsungHero97) [inline src/views/Main/components/Navbar.js]: nice :)

File: src/views/Main/components/styles.module.css [hunk 5]
```diff
@@ -102,13 +102,17 @@
 }
 
 .navButton {
+  display: flex;
```
Comment (UnsungHero97) [inline src/views/Main/components/styles.module.css]: nice :)

File: postcss.config.js (hunk 1 above)
Comment (mindblight) [inline postcss.config.js]: Deleted. It's a leftover from debugging

File: postcss.config.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
```
Comment (UnsungHero97) [inline postcss.config.js]: As much as possible, I'd like to keep all such configs inside the config directory. Can you try this... move this config to `config/postcss/postcss.config.js` instead, and update webpack with the following:

    loader: ExtractTextPlugin.extract('css-loader?modules&importLoader=1!postcss-loader?config=../postcss/postcss.config.js'),

https://github.com/postcss/postcss-loader/pull/143

File: src/containers/catalog/CatalogComponent.js (hunk 2 above)
Comment (mindblight) [inline src/containers/catalog/CatalogComponent.js]: I don't have a strong opinion about one over the other. I followed the pattern in the current code, but they should transpile to the same thing. 

Typescript lets you define an interface for `props`, so you don't use `propTypes` at all. If Flow has something similar, I'd love to use that :)

File: postcss.config.js (hunk 6 above)
Comment (mindblight) [inline postcss.config.js]: 👍  much better. Done

File: src/views/Main/components/styles.module.css (hunk 5 above)
Comment (mindblight) [inline src/views/Main/components/styles.module.css]: We'll have to be a little careful with IE11, but it looks manageable :). http://caniuse.com/#feat=flexbox

**Pull request merged**